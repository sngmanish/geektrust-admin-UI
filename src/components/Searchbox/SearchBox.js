import React from "react";

const SearchBox = ({ searchText, setSearchText }) => {
  return (
    <div
      style={{
        width: "100%",
        borderBottom: "1px solid rgb(158, 155, 155)",
        height: "40px",
        // ":hover": { borderBottom: "1px solid blue" },
      }}
    >
      <input
        name="search"
        value={searchText}
        onChange={(event) => setSearchText(event.target.value)}
        placeholder="Search by name, email or role"
        style={{
          width: "100%",
          boxSizing: "border-box",
          padding: "10px",
          height: "40px",
          backgroundColor: "transparent",
          border: "none",
          outline: "none",
        }}
      />
    </div>
  );
};

export default SearchBox;
