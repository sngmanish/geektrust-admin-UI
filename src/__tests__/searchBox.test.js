import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import SearchBox from '../components/SearchBox/SearchBox'

describe('SearchBox', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SearchBox searchText="" setSearchText={() => {}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('updates search text on input change', () => {
    const setSearchText = jest.fn()
    render(<SearchBox searchText="" setSearchText={setSearchText} />)
    const input = screen.getByTestId('search-box-input')
    fireEvent.change(input, { target: { value: 'test' } })
    expect(setSearchText).toHaveBeenCalledWith('test')
  })
})
