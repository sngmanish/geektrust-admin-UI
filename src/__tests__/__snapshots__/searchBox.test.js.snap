// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correctly 1`] = `
<div
  style={
    Object {
      "borderBottom": "1px solid rgb(158, 155, 155)",
      "height": "40px",
      "width": "100%",
    }
  }
>
  <input
    data-testid="search-box-input"
    name="search"
    onChange={[Function]}
    placeholder="Search by name, email or role"
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "none",
        "boxSizing": "border-box",
        "height": "40px",
        "outline": "none",
        "padding": "10px",
        "width": "100%",
      }
    }
    value=""
  />
</div>
`;
